.github/workflows/release-public-artifacts.yml

```diff
@@ -0,0 +1,95 @@
+name: Publish public artifacts
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: The tag to build from.
+        required: true
+
+concurrency:
+  # This only runs on master
+  group: ci-${{ github.run_id || github.ref_name }}
+  cancel-in-progress: true
+
+jobs:
+  setup_x86:
+    uses: ./.github/workflows/setup-runner.yml
+    with:
+      username: master
+      runner_type: builder-x86
+    secrets: inherit
+
+  setup_arm:
+    uses: ./.github/workflows/setup-runner.yml
+    with:
+      username: master
+      runner_type: builder-arm
+    secrets: inherit
+
+  build:
+    strategy:
+      matrix:
+        include:
+          - arch: x86_64
+            runner-label: master-x86
+          - arch: arm64
+            runner-label: master-arm
+    runs-on: ${{ matrix.runner-label }}
+    steps:
+      - uses: actions/checkout@v4
+        with: { ref: "${{ env.GIT_COMMIT }}" }
+      - uses: ./.github/ci-setup-action
+        with:
+          dockerhub_password: "${{ secrets.DOCKERHUB_PASSWORD }}"
+          concurrency_key: build-release-artifacts-${{ github.actor }}-${{ matrix.arch }}
+      - name: "Build images"
+        timeout-minutes: 40
+        run: |
+          earthly-ci --no-output --build-arg DIST_TAG=latest --build-arg ARCH=${{ matrix.arch }} ./yarn-project+export-aztec-arch
+          earthly-ci --no-output --build-arg DIST_TAG=latest --build-arg ARCH=${{ matrix.arch }} ./noir+export-noir
+
+  publish:
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.ref }}
+      - uses: ./.github/ci-setup-action
+        with:
```

Comment:
this step needs a strategy and runner_type in, then the rest can be run-on-builder instead of ensure-builder (it's a bad name, it used to ensure it but now is really start-builder)